test/pub_update_test.dart

```diff
@@ -129,6 +130,56 @@ void main() {
         );
       });
 
+      test(
+        '''returns true when currentVersion is pre release and latestVersion is stable''',
+        () async {
+          expect(
+            await pubUpdater.isUpToDate(
+              packageName: 'very_good_cli',
+              currentVersion: '0.4.0-dev.1',
+            ),
+            true,
+          );
+        },
+      );
+
+      group('pre releases', () {
+        setUp(() {
+          client = MockClient();
+          response = MockResponse();
+          pubUpdater = PubUpdater(client);
+          processManager = MockProcessManager();
+
+          when(() => client.get(any())).thenAnswer((_) async => response);
+          when(() => response.statusCode).thenReturn(HttpStatus.ok);
+          when(() => response.body)
+              .thenReturn(preReleasePackageInfoResponseBody);
+
+          when(() => processManager.run(any()))
+              .thenAnswer((_) => Future.value(FakeProcessResult()));
+        });
+
+        test('returns false when currentVersion < latestVersion', () async {
```

Comment:
So just to clarify if the current version was `0.1.0-dev.48` and the latest version was 1.0.0, `isUpToDate` would return `true`?

Reply:
I would say, in short, I think we shouldn't track how "up to date" a pre-release is when compared to a semantic release because it is a rabbit hole (https://github.com/VeryGoodOpenSource/pub_updater/issues/41)